depthai_nodes/node/tiles_patcher.py

```diff
@@ -220,34 +410,52 @@ def _send_output(
         device_timestamp: datetime.timedelta,
         transformation: Optional[dai.ImgTransformation],
         sequence_num: int,
+        input_type: Type[Union[dai.ImgDetections, ImgDetectionsExtended]],
     ) -> None:
         """Send the final combined bounding boxes as output when all tiles for a frame
         are processed.
 
         @param timestamp: The timestamp of the frame.
         @param device_timestamp: The timestamp of the frame on the device.
         """
-        combined_bboxes: List[dai.ImgDetection] = []
+        combined_bboxes: Union[List[dai.ImgDetection], List[ImgDetectionExtended]] = []
         for bboxes in self.tile_buffer:
             combined_bboxes.extend(bboxes)
 
-        if combined_bboxes:
+        if combined_bboxes and input_type == dai.ImgDetections:
```

Comment:
I don't think we need to differentiate between `combined_bboxes and input_type == dai.ImgDetections` and `not combined_bboxes and input_type == dai.ImgDetections`.
Because if `combined_bboxes` is empty list then `nms_detections` returns empty list. And so to me it looks like doing just:
```
if input_type == dai.ImgDetections:
    detection_list = nms_detections(
                combined_bboxes,
                conf_thresh=self.conf_thresh,
                iou_thresh=self.iou_thresh,
            )
            detections = dai.ImgDetections()
            detections.detections = detection_list
```
will always work